cmd/katapult/networks_test.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/krystal/go-katapult/core"
+	"github.com/krystal/katapult-cli/config"
+	"github.com/stretchr/testify/assert"
+)
+
+var idNetworks = []*core.Network{
+	{
+		ID:        "pognet",
+		Name:      "Pognet 1",
+		Permalink: "pog-1",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+	{
+		ID:        "pognet2",
+		Name:      "Pognet 2",
+		Permalink: "pog-2",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+}
+
+var subdomainNetworks = []*core.Network{
+	{
+		ID:        "pognet3",
+		Name:      "Pognet 3",
+		Permalink: "pog-3",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+	{
+		ID:        "pognet4",
+		Name:      "Pognet 4",
+		Permalink: "pog-4",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+}
+
+func mockNetworksServer() (net.Listener, error) {
+	ln, err := net.Listen("tcp", "127.0.0.1:7654")
```

Comment:
If you are going to use local http server rather than mocking with an interface, you should look at https://golang.org/pkg/net/http/httptest/#NewServer . This gives you a random port, which means we can run the tests concurrently.

You can see some prior art here on one of my repos: https://github.com/zmb3/spotify/blob/master/spotify_test.go#L12